python/gui/static/js/utilities.js

```diff
@@ -292,6 +292,118 @@ export function downloadData(name, data) {
     }, 0);
 }
 
+export async function getServerData(url, dataType) {
+    if (dataType == null) {
+        dataType = "json";
+    }
+
+    dataType = dataType.toLowerCase();
+
+    const validDataTypes = ["blob", "json", "text", "buffer"];
+
+    if (!validDataTypes.includes(dataType)) {
+        throw new Error(
+            `"${dataType}" is not a valid data type for server retrieval. Valid options are: ${validDataTypes.join(", ")}`
+        );
+    }
+
+    return await fetch(url)
+        .then(
+            (response) => {
+                if (dataType === "blob") {
+                    return response.blob();
+                }
+                else if (dataType === "json") {
+                    return response.json();
+                }
+                else if (dataType === "buffer") {
+                    return response.arrayBuffer();
+                }
+                return response.text();
+            }
+        );
+}
+
+/**
+ * Attach dynamic code to the document
+ *
+ * Dynamic code may come from a request of some sort
+ *
+ * WARNING: Be careful about what is calling this; there are no steps to protect the browser from malicious code
+ *
+ * @param {string} code The code to add
+ * @param {string?} languageMIMEType What type of object to convert the following blob into. 'application/javascript' is fine 99% of the time.
+ */
+export function attachCode(code, languageMIMEType) {
+    console.warn("Attaching dynamic code - be sure that the loaded code is safe to use");
+
+    if (languageMIMEType == null || typeof languageMIMEType !== 'string' || languageMIMEType === '') {
+        languageMIMEType = "application/javascript";
+    }
+
+    let scriptID = Math.floor(Math.random() * 1000).toString();
```

Comment:
I just decided to use this to come up with some random string. The value isn't important - just a random id to put on a `<script>` element. Using that function maybe once is a tad questionable, using multiple times even more so, since that function literally takes arbitrary code and loads it into the document for use.

Should I replace that with `crypto.randomUUID()`?